src/concepts/rendering-under-the-hood.md

```diff
@@ -0,0 +1,265 @@
+    Widget <|.. Table
+
+
+```
+
+Basically what this all means that any content rendered to a `Buffer` is only stored in that
+particular `Buffer` struct in the form of `Cell`s that contain string and style information.
+
+```admonish note
+Because any styling has to happen though the `Style` struct (i.e. methods that apply
+`Style`s to `Cell`s) you can't use ansi escape sequences directly in the string content.
+
+As a workaround, what you can do however is convert the ansi escape sequences into a
+`ratatui::text::Text` struct that contains the appropriate `Style`s. There's a crate for this that
+might help: <https://crates.io/crates/ansi-to-tui>. And this `Text` struct that you create can be
+passed into the `Paragraph` widget to get the desired effect.
+
+Even if this particular crate doesn't work for you this approach is probably what you want to use if
+you want to render ansi escape sequences in Ratatui.
+You can learn more about the text related of Ratatui features and displaying text
+[here](./../how-to/render/display-text.md).
+```
+
+After all the `Widget`s are rendered to the same `Buffer` struct, i.e. after
+`terminal.draw(|f| ...)` returns, [`terminal.flush()`] is called. This is where the content is
+actually written out to the terminal.
+
+In `flush()`, the difference between the previous and the current buffer is obtained and that is
+passed it to the current backend for drawing to your actual terminal.
+
+```rust
+    pub fn flush(&mut self) -> io::Result<()> {
+        let previous_buffer = &self.buffers[1 - self.current];
+        let current_buffer = &self.buffers[self.current];
+        let updates = previous_buffer.diff(current_buffer);
+        if let Some((col, row, _)) = updates.last() {
+            self.last_known_cursor_pos = (*col, *row);
+        }
+        self.backend.draw(updates.into_iter())
+    }
+```
+
+After `flush()`, the buffers are swapped and the next time `terminal.draw(|f| ...)` is called a
+`Frame` is constructed with the other `Buffer` struct. This is why it is called a double-buffer
+rendering technique.
+
+The important thing to note is that because all widgets render to the same `Buffer` within a single
+`terminal.draw(|f| ...)` call, rendering of different widgets may overwrite the same `Cell`, so the
+order in which widgets are rendered is relevant. For example, in this `draw` example below,
+`"content1"` will be overwritten by `"content2"` which will be overwritten by `"content3"` in the
+`Buffer` struct, and Ratatui will only ever write out `"content3"` to the terminal:
+
+```rust
+terminal.draw(|f| {
+    f.render_widget(Paragraph::new("content1"), f.size());
+    f.render_widget(Paragraph::new("content2"), f.size());
+    f.render_widget(Paragraph::new("content3"), f.size());
+})
+```
+
+This was a sneak peak into what happens under the hood in Ratatui. If you have any questions, feel
+free to open issues or discussions on [the main repository](https://github.com/ratatui-org/ratatui).
```

Comment:
In summary, the application calls draw to ..., and then blah blah blah. Each widget does blah blah blah.... Once the user inteface closure completes, ratatui ...

Perhaps drop the questions, etc. part (we don't need to repeat that on every page or if we do, then just having a common snippet would be better.

Reply:
https://github.com/ratatui-org/ratatui-book/pull/99/commits/ef16b8baf97267e8e4fb1e836599165995415beb